docs/developer/local_up_kubeedge.md

```diff
@@ -0,0 +1,173 @@
+
+1. Preparation:
+Ensure that you have kind, kubectl, docker, and other necessary tools installed.
+Ensure that your system has a Go language development environment installed, and the version meets the requirements of KubeEdge.
+Based on the user's choice of container runtime (docker, cri-o, isulad), user can verify the prerequisites via the script below, or it will also be verified before the installation start in **hack/local_up_kubeedge.sh**
+
+```shell
+verify_docker_installed
+verify_crio_installed
+verify_isulad_installed
+kubeedge::golang::verify_golang_version
+check_kubectl
+check_kind
+```
+
+2. Executing the Script:
+Before running the script, you can modify some environment variables to suit your needs, such as **LOG_DIR** and **CONTAINER_RUNTIME** in the script.
+The script should be run with administrative privileges, as some operations (e.g., software installation, system configuration changes) require sudo access.
+
+```shell
+hack/local_up_kubeedge.sh
+```
+This script will run the environment setting, prerequesites verification, installation and so on. For details, please refer [Script Analysis](#script-analysis)
+
+3. Checking the Running Status:
+After the script completes, you can use kubectl commands to check the status of the cluster's nodes and the running status of KubeEdge components.
+
+```shell
+healthcheck
+kubectl get nodes | grep edge-node | grep -q -w Ready && break
+```
+
+4. Debugging and Log Viewing:
+If problems occur, you can view the log files generated by the script, typically located at 
+
+```shell
+/tmp/cloudcore.log and
+/tmp/edgecore.log
+```
+
+5. Shutting Down the Cluster:
+If the cluster is no longer needed, you can use the cleanup function provided in the script to stop and clean up the cluster and related resources.
+
+```shell
+cleanup
+```
+
+## Script Analysis
+
+local-up-kubeedge.sh scipt is a tool for installing and configuring a KubeEdge environment, below are some main features it provides:
+
+1. Set environment variables
+
+This script sets several environment variables at the beginning, including log directory, log level, timeout duration, protocol type, container runtime type, Kebenetes cluster name and etc. This is the necessary for local Kebeedge cluster setting up and starting.
+
+```shell
+KUBEEDGE_ROOT=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)/..
+ENABLE_DAEMON=${ENABLE_DAEMON:-false}
+LOG_DIR=${LOG_DIR:-"/tmp"}
+LOG_LEVEL=${LOG_LEVEL:-2}
+TIMEOUT=${TIMEOUT:-60}s
+PROTOCOL=${PROTOCOL:-"WebSocket"}
+CONTAINER_RUNTIME=${CONTAINER_RUNTIME:-"containerd"}
+KIND_IMAGE=${1:-"kindest/node:v1.29.0"}
+```
+
+2. Clean up the previous installation
+
+```shell
+cleanup
+```
+It will uninstall kubeedge and the cluster.
+
+
+3. Container Runtime Installation:
+
+Based on the user's choice of container runtime (docker, cri-o, isulad), the script attempts to install the corresponding runtime tools.
+Checks and verifies the installation of Go language version, kubectl tool, and kind tool.
```

Comment:
Would it be better to mention here that if the user does not set the container runtime, we will use containerd by default?

Reply:
Add note that containerd will be set by default.